lightning/backend/sql2kv.go

```diff
@@ -332,26 +334,32 @@ func (kvcodec *tableKVEncoder) Encode(
 		record = append(record, value)
 
 		if isAutoRandom && isPk {
-			incrementalBits := autoRandomIncrementBits(col, int(kvcodec.tbl.Meta().AutoRandomBits))
+			incrementalBits := autoRandomIncrementBits(col, int(meta.AutoRandomBits))
 			kvcodec.tbl.RebaseAutoID(kvcodec.se, value.GetInt64()&((1<<incrementalBits)-1), false, autoid.AutoRandomType)
 		}
 		if isAutoIncCol {
 			kvcodec.tbl.RebaseAutoID(kvcodec.se, value.GetInt64(), false, autoid.AutoIncrementType)
 		}
 	}
 
-	if common.TableHasAutoRowID(kvcodec.tbl.Meta()) {
+	if common.TableHasAutoRowID(meta) {
+		rowValue := rowID
 		j := columnPermutation[len(cols)]
 		if j >= 0 && j < len(row) {
 			value, err = table.CastValue(kvcodec.se, row[j], extraHandleColumnInfo, false, false)
+			rowValue = value.GetInt64()
 		} else {
+			if meta.ShardRowIDBits > 0 {
+				shard := (int64(fastrand.Uint32()) & (1<<meta.ShardRowIDBits - 1)) << (autoid.RowIDBitLength - meta.ShardRowIDBits - 1)
```

Comment:
using `fastrand.Uint32()` means the row ID is not reproducible with checkpoint. please use a deterministic seed like `autoRandomBits`.